optimism/src/mips/constraints.rs

```diff
@@ -440,10 +450,173 @@ impl<Fp: Field> InterpreterEnv for Env<Fp> {
         _len: &Self::Variable,
         pos: Self::Position,
     ) -> Self::Variable {
-        Expr::Atom(ExprInner::Cell(Variable {
+        // The (at most) 4-byte chunk that has been read from the preimage
```

Comment:
1. The first 8 bytes are the length of the total preimage.
2. We have to write into the memory on the constraint side. On the witness side, we do update the linear memory. However, on the constraint side, we must add a lookup into `MemoryLookup`.

Can be done in a follow-up.